kong/status/ready.lua

```diff
@@ -0,0 +1,62 @@
+local declarative = require "kong.db.declarative"
+local constants = require "kong.constants"
+
+local tonumber = tonumber
+local kong = kong
+
+local get_current_hash = declarative.get_current_hash
+
+
+local worker_count = ngx.worker.count()
+local kong_shm = ngx.shared.kong
+local is_traditional = kong.configuration.database ~= "off"
+
+local DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY = 
+                                constants.DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY
+local DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY =
+                                constants.DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY
+local DECLARATIVE_EMPTY_CONFIG_HASH = constants.DECLARATIVE_EMPTY_CONFIG_HASH
+
+local function is_ready()
+
+  local ok = kong.db:connect()
+  if not ok then
+    return false
+  end
+
+  if is_traditional then
+    kong.db:close() -- ignore ERRs
+    return true
+  end
+
+  local router_rebuilds = 
+      tonumber(kong_shm:get(DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY)) or 0
+  local plugins_iterator_rebuilds = 
+      tonumber(kong_shm:get(DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY)) or 0
+
+  if router_rebuilds < worker_count 
+    or plugins_iterator_rebuilds < worker_count
+  then
+    return false
+  end
+
+  local current_hash = get_current_hash()
+
+  if not current_hash or current_hash == DECLARATIVE_EMPTY_CONFIG_HASH then
+    return false
+  end
+
+  return true
```

Comment:
Please change the function signature to respond back with an error as well.
Please include a descriptive error in here that can be returned as part of the JSON body of 503 response when Kong is not ready.
This helps communicate back to the user (and debug Kong) as to which data-structure is not ready and what is the issue in Kong.

Reply:
> Returning an error is fine, but I don't think we should display the error message to the user because the error information could be sensitive. Even if the error message consists of some simple descriptive words.